DESIGN-PROPOSALS.md

```diff
@@ -0,0 +1,93 @@
+# Design Proposal Template
+
+**Authors**: alice alice@example.com, bob jones [bob@example.com](mailto:bob@example.com)
+
+**Begin Design Discussion**: YYYY-MM-DD
+
+**(optional) Status: **implementable, accepted, deferred, rejected, withdrawn, or replaced
+
+**Checklist**:
+
+- [ ] Prerequisite 1
+- [ ] Prerequisite 2
+- [ ] Component 1
+- [ ] Component 2
+- [ ] Docs
+- [ ] Tests
```

Comment:
Should these "prerequisite" and "component" lines be something else?
I read @jberkus's comment to suggest that this checklist be all of the phases in the lifecycle of a design proposal, and they get checked off as the proposal moves through those phases. I assume that "prerequisite 1"... "Component 2" aren't phases in the lifecycle.

Reply:
If the idea is for the authors to fill this checklist with pieces of the proposal and not lifecycle stages of a proposal, I think that we need to make that clear with a sentence or two above the checklist explaining how an author will use it.

Reply:
It's supposed to be a generic "There should be a checklist here".  What goes in the checklist is highly dependent on how the individual project manages their PRs.

But regardless, I think we're getting to the point of "we can't move foward on this without also having documentation for the template user".